src/Peering/Peering/PeeringService/Prefix/DeleteAzurePeeringServicePrefixCommand.cs

```diff
@@ -0,0 +1,165 @@
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <summary>
+    /// New Azure InputObject Command-let
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzPeeringServicePrefix", DefaultParameterSetName = Constants.ParameterSetNameByName, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class DeleteAzurePeeringServicePrefixCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the legacy InputObject.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = Constants.ParameterSetNameDefault,
+            HelpMessage = Constants.PrefixInputObjectHelp + "Prefix")]
+        [ValidateNotNullOrEmpty]
+        public PSPeeringServicePrefix InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets The Resource Group Name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets The InputObject NameMD5AuthenticationKeyHelp
+        /// </summary>
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ValidateNotNullOrEmpty]
+        public string PeeringServiceName { get; set; }
+
+        /// <summary>
+        /// The resource  id
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceIdHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.ForceHelp)]
+        public SwitchParameter Force { get; set; }
```

Comment:
@rileymckenna is this `-Force` parameter is needed? Typically, cmdlets have a `-Force` parameter whenever they have an additional prompt to the user that they want to allow them to skip. For example, if this cmdlet is deleting more than just the prefix, we should prompt the user to let them know what other resources are being deleted.

Reply:
So this is a tough one. I might be thinking about this wrong, but what if i wanted to do a -WhatIf and accidentally delete my resource. I know Typing out the command is one things, but i feel like a Delete situation should have a warning prompt "Hey you are about to remove this resource". The convention in portal is do a warning when delete and users have to type something out in order to delete a resource. We just do a y or n. So i am going to leave it unless its strongly discouraged.